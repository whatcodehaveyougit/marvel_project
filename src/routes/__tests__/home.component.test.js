import { render, cleanup, fireEvent, screen } from '@testing-library/react'
import { MemoryRouter as Router } from "react-router-dom";
import Home from '../home/home.component'

afterEach(cleanup)

const charactersData =   [
  {
    id: '890',
    name: 'The Man',
    thumbnail: {
      path: 'this-it-the-path',
      extension: 'jpg'
    },
    description: 'He is a big monstor'
  },
  {
    id: '123',
    name: 'The Hulk',
    thumbnail: {
      path: 'this-it-the-path2',
      extension: 'jpg'
    },
    description: 'He is a big monstor'
  }
]

const MockHome = () => {
  return (
    <Router>
      <Home
        charactersData={charactersData}
      />
    </Router>
  )
}

describe('HomeComponent Tests', function() {

  it('Search input loads', () => {
    render( <MockHome /> )
    const inputElement = screen.getByPlaceholderText(/Search Marvel Characters.../)
    expect(inputElement).toBeInTheDocument()
  })

  it('We can type into input field', () => {
    render( <MockHome /> )
    const inputElement = screen.getByPlaceholderText(/Search Marvel Characters.../)
    fireEvent.change(inputElement, { target: { value: 'Hulk' } } )
    expect(inputElement.value).toBe('Hulk')
  })

  it('HomeComponent matches snapshot', () => {
    const { container } = render ( <MockHome /> )
    expect(container).toMatchSnapshot()
  })


})