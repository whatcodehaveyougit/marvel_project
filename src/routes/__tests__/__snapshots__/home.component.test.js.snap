// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent Tests HomeComponent matches snapshot 1`] = `
<div>
  <div
    class="home-container"
  >
    <div
      class="search-bar-component"
    >
      <input
        class="search-monsters-input"
        name="search-monsters"
        placeholder="Search Marvel Characters..."
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 character-card-container css-11l5t4l-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root character-card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <img
            alt="The Man"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img thumbnail-image css-o69gx8-MuiCardMedia-root"
            src="this-it-the-path.jpg"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              The Man
            </h5>
            <div
              class="MuiGrid-root max-lines css-vj1n65-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              >
                He is a big monstor
              </p>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing card-actions css-1t6e9jv-MuiCardActions-root"
          >
            <a
              class="character-page-link"
              href="/character/890"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Find out more
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 character-card-container css-11l5t4l-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root character-card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <img
            alt="The Hulk"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img thumbnail-image css-o69gx8-MuiCardMedia-root"
            src="this-it-the-path2.jpg"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              The Hulk
            </h5>
            <div
              class="MuiGrid-root max-lines css-vj1n65-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              >
                He is a big monstor
              </p>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing card-actions css-1t6e9jv-MuiCardActions-root"
          >
            <a
              class="character-page-link"
              href="/character/123"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Find out more
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
